### PR TITLE
[Gutenberg] Upload React Native bundle and source map files from Gutenberg framework

### PR DESCRIPTION
**Pull request by fluiddot**
**Related PRs:**
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6760

**To test:**
1. Run command: `bundle exec pod install`.
2. Observe that the source map file (`App.composed.js.map` or `main.jsbundle.map`) is not present in the folder `Pods/Gutenberg/Frameworks/Gutenberg.xcframework/ios-arm64/Gutenberg.framework`.
4. Download the installable build and inspect the IPA file.
5. Observe that the source map file (`App.composed.js.map` or `main.jsbundle.map`) is not present, only the file `main.jsbundle`.
6. Observe in the CI job `Jetpack Prototype Build` or `WordPress Prototype Build`, that the step `sentry_upload_sourcemap ` uploads the bundle and source map files to Sentry.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. 
What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/lanes/build.rb [hunk 1]
```diff
@@ -532,39 +532,31 @@ def upload_build_to_app_center(
...
-        app_identifier:,
-        # When the React native bundle is generated, the source map file references
-        # include the local machine path, with the `rewrite` and `strip_common_prefix`
-        # options Sentry automatically strips this part.
-        rewrite: true,
-        strip_common_prefix: true,
-        sourcemap: sourcemaps_folder
-      )
-    end
+    gutenberg_bundle_source_map_folder = File.join(PROJECT_ROOT_FOLDER, 'Pods', 'Gutenberg', 'react-native-bundle-source-map')
```
Comment (fluiddot) [inline fastlane/lanes/build.rb]: This is now the path where the bundle and source map files are located for uploading to Sentry ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/296dd56d4d4761a85d4f587f3cb72e7d5e39e692/ios-xcframework/build.sh#L172-L176)).

File: fastlane/lanes/build.rb [hunk 2]
```diff
@@ -532,39 +532,31 @@ def upload_build_to_app_center(
   end
 
   def upload_gutenberg_sourcemaps(sentry_project_slug:, release_version:, build_version:, app_identifier:)
-    # The bundle and source map files are the same for all architectures.
-    gutenberg_bundle = File.join(PROJECT_ROOT_FOLDER, 'Pods/Gutenberg/Frameworks/Gutenberg.xcframework/ios-arm64/Gutenberg.framework')
-
-    Dir.mktmpdir do |sourcemaps_folder|
```
Comment (fluiddot) [inline fastlane/lanes/build.rb]: We no longer need to create a temporary path because the files are already located in a separate folder.

**Review by jhnstn — APPROVED** [review]
Works as expected. Nice work!

**Pull request merged**